Source/SpatialGDK/Private/SpatialInterop.cpp

```diff
@@ -860,16 +866,136 @@ bool USpatialInterop::IsSingletonClass(UClass* Class)
 	return false;
 }
 
-NameToEntityIdMap* USpatialInterop::GetSingletonNameToEntityId() const
+improbable::unreal::GlobalStateManagerData* USpatialInterop::GetGlobalStateManagerData() const
 {
 	TSharedPtr<worker::View> View = SpatialOSInstance->GetView().Pin();
 	auto It = View->Entities.find(SpatialConstants::GLOBAL_STATE_MANAGER);
 
 	if (It != View->Entities.end())
 	{
 		improbable::unreal::GlobalStateManagerData* GSM = It->second.Get<improbable::unreal::GlobalStateManager>().data();
-		return &(GSM->singleton_name_to_entity_id());
+		return GSM;
 	}
 
 	return nullptr;
 }
+
+NameToEntityIdMap* USpatialInterop::GetSingletonNameToEntityId() const
+{
+	improbable::unreal::GlobalStateManagerData* GSM = GetGlobalStateManagerData();
+
+	if (GSM == nullptr) return nullptr;
+
+	return &(GSM->singleton_name_to_entity_id());
+}
+
+EntityIdToPathMap* USpatialInterop::GetEntityIdToReplicatedStablyNamedPath() const
+{
+	improbable::unreal::GlobalStateManagerData* GSM = GetGlobalStateManagerData();
+
+	if (GSM == nullptr) return nullptr;
+
+	return &(GSM->entity_id_to_replicated_stably_named_path());
+}
+
+void USpatialInterop::ReserveReplicatedStablyNamedActorChannel(USpatialActorChannel* Channel)
+{
+	ReplicatedStablyNamedActorChannelQueue.Add(Channel);
+
+	if (bCanSpawnReplicatedStablyNamedActors)
+	{
+		ReserveReplicatedStablyNamedActors();
+	}
+}
+
+void USpatialInterop::UnreserveReplicatedStablyNamedActor(AActor* Actor)
+{
+	// Slightly inefficient linear search, could use a hashmap
+	USpatialActorChannel* Channel = nullptr;
+	for (auto& It : ReplicatedStablyNamedActorChannelQueue)
+	{
+		if (It->Actor == Actor) {
+			Channel = It;
+			break;
+		}
+	}
+
+	if (Channel != nullptr) {
+		ReplicatedStablyNamedActorChannelQueue.Remove(Channel);
+	}
+}
+
+void USpatialInterop::AddReplicatedStablyNamedActorToGSM(const FEntityId& EntityId, AActor* Actor)
+{
+	EntityIdToPathMap& EntityIdToReplicatedStablyNamedPathMap = *GetEntityIdToReplicatedStablyNamedPath();
```

Comment:
You're potentially dereferencing a `nullptr` here! Looks a bit off anyway, if you're returning a pointer from the function why don't you just use it as a pointer and null check as needed?

Reply:
Yes, I tried to stick with the previous similar function ```USpatialInterop::ExecuteInitialSingletonActorReplication(const NameToEntityIdMap& SingletonNameToEntityId)```, which has the map in the arguments and also ```USpatialInterop::UpdateGlobalStateManager```, which does 
```NameToEntityIdMap& SingletonNameToEntityId = *GetSingletonNameToEntityId();```
but UpdateGlobalStateManager should probably be changed either way to something like AddSingletonToGSM.